File: examples/standalone_cpp/Limits/python/CPULimit.py
#CPULimit.py
from riaps.run.comp import Component
import os
import logging

class CPULimit(Component):
    def __init__(self):
        super(CPULimit, self).__init__()	        
        self.pid = os.getpid()
        self.logger.info("(PID %s) - starting CPULimit, %s",str(self.pid),str(now))
        
    def on_ticker(self):
        now = self.ticker.recv_pyobj()
        self.logger.info('PID(%s) - on_ticker(): %s',str(self.pid),str(now))
    
	def __destroy__(self):			
		self.logger.info("(PID %s) - stopping CPULimit, %s",str(self.pid),now)   	        	        

File: examples/standalone_cpp/Limits/cpp/CPULimit.cc
#include "include/CPULimit.h"

namespace limits {
   namespace components {
      
      CPULimit::CPULimit(_component_conf &config, riaps::Actor &actor) :
      CPULimitBase(config, actor) {
      }
      
      void CPULimit::OnTicker(riaps::ports::PortBase *port) {
        _logger->info("{}", __PRETTY_FUNCTION__);
      }
      
      void CPULimit::OnGroupMessage(const riaps::groups::GroupId& groupId,
      capnp::FlatArrayMessageReader& capnpreader, riaps::ports::PortBase* port){
         
      }

       void CPULimit::handleCPULimit(){
           _logger->info("{} is called", __PRETTY_FUNCTION__);
       }
      
      CPULimit::~CPULimit() {
         
      }
   }
}

riaps::ComponentBase *create_component(_component_conf &config, riaps::Actor &actor) {
   auto result = new limits::components::CPULimit(config, actor);
   return result;
}

void destroy_component(riaps::ComponentBase *comp) {
   delete comp;
}
